lib/manager/composer/artifacts.ts

```diff
@@ -119,6 +117,20 @@ export async function updateArtifacts({
         tagScheme: composerVersioningId,
       },
     };
+
+    // Determine whether Symfony Flex has been installed
+    const hasSymfonyFlex =
+      existingLockFile.packages?.some((p) => p.name === 'symfony/flex') ||
+      existingLockFile['packages-dev']?.some((p) => p.name === 'symfony/flex');
```

Comment:
Can you move this to a util function and make it more generic, so it's easily extendable to more packages in future if requirewd?

Reply:
Something like this?

```js
export function hasComposerDependencyInstalled(
  lock: ComposerLock,
  dependency: string,
  includeDev?: boolean,
): boolean {
  if (lock.packages?.some((p) => p.name === dependency)) {
    return true;
  }

  return includeDev && lock['packages-dev']?.some((p) => p.name === dependency);
}
```